cpp/purify/CMakeLists.txt

```diff
@@ -67,25 +65,30 @@ if(PURIFY_ARRAYFIRE)
   target_link_libraries(libpurify ${ArrayFire_LIBRARIES})
 endif()
 
-target_link_libraries(libpurify ${CONAN_LIBS})
-
 target_include_directories(libpurify PUBLIC
   $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/cpp>
   $<INSTALL_INTERFACE:include/>)
 
 
 add_include_dir(
-  ${EIGEN3_INCLUDE_DIR}
+  ${Eigen3_INCLUDE_DIR}
+  ${eigen_INCLUDE_DIR}
   ${Boost_INCLUDE_DIR}
-  ${CFitsIO_INCLUDE_DIR}
-  ${Yamlcpp_INCLUDE_DIR}
+  ${cfitsio_INCLUDE_DIR}
+  ${yaml-cpp_INCLUDE_DIR}
   ${Cubature_INCLUDE_DIR}
   ${CImg_INCLUDE_DIR}
-)
+  ${sopt_INCLUDE_DIR}
+  )
 target_link_libraries(libpurify
-  ${FFTW3_DOUBLE_LIBRARY}  ${CFitsIO_LIBRARY} ${Sopt_CPP_LIBRARY} ${X11_X11_LIB} ${Yamlcpp_LIBRARY} ${Cubature_LIBRARIES}
-  ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY}
+  ${Cubature_LIBRARIES}
+  FFTW3::fftw3
+  cfitsio::cfitsio
+  sopt::sopt
+  yaml-cpp
+  Boost::filesystem
```

Comment:
We had a [previous problem](https://github.com/astro-informatics/purify/pull/302#issuecomment-1085669536) with Boost missing components, if `fiber` included in these?